plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -418,23 +424,60 @@ int hash_rules(const std::vector<std::string> &irbs, const int pid, std::string
     return 0;
 }
 
-class make_copy {
-public:
-        make_copy(const std::vector<std::string>& _irbs, const int _pid) : irbs_(_irbs), pid_(_pid) {
-            for(auto const &irb : _irbs) {
-                boost::filesystem::copy_file(get_rule_base_path(irb), get_rule_base_path_copy(irb, _pid));
-            }
-        }
-        ~make_copy() {
-            for(auto const &irb : irbs_) {
-                boost::filesystem::remove(get_rule_base_path_copy(irb, pid_));
+class in_memory_rulebases
+{
+  public:
+    in_memory_rulebases(const std::vector<std::string>& _irods_rule_bases)
+    {
+        for (auto const& irb : _irods_rule_bases) {
+            std::ifstream ifs(get_rule_base_path(irb));
+            if (!ifs) {
+                log_re::warn(
+                    "in_memory_rulebases: input file stream [{}] failed, errno = {}", get_rule_base_path(irb), errno);
             }
+            rule_bases_.insert(
+                {irb, std::string({std::istreambuf_iterator<char>(ifs), std::istreambuf_iterator<char>()})});
```

Comment:
What is the outcome of the istreambuf iterators (on L439) when `ifs` is in a bad state?

Reply:
Oops, good catch. Fixing...